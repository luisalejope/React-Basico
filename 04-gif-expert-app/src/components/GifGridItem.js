import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({id, title, url}) => {
    return (
        <>
        <img src={url} alt={title}></img>
        </>
    )
}

GifGridItem.propTypes = {

}

export default GifGridItem
