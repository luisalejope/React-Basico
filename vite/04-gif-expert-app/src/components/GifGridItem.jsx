import PropTypes from 'prop-types'

export const GifGridItem = ({title, url, id}) => {
  return (
    <div className='card' id={id}>
        <h2>
            {title.trim() !== '' ? title : 'GIF SIN TITULO'}
        </h2>
        <img src={url} alt={title} />
    </div>
  )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string
}
