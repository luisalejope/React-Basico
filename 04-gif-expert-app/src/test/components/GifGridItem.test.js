import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas con <GifGridItem/>", () => {
  const title = "Prueba titulo";
  const url = "http://localhot/algo.jpg";
  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  beforeEach(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });

  test("debe de mostrar <GifGridItem/>", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener un parrafo con el title", () => {
    const titulo = wrapper.find("p").text().trim();
    expect(titulo).toBe(title);
  });

  test("debe de obtener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img").props();
    expect(img.src).toBe(url);
    expect(img.alt).toBe(title);
  });

  test("debe de tener dentro del className animate__backInLeft", () => {
    const div = wrapper.find("div").props();
    const className = div.className;
    expect(className.includes("animate__backInLeft")).toBe(true);
  });
});
