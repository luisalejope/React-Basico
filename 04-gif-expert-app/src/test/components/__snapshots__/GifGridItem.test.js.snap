// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifGridItem/> debe de mostrar <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="Prueba titulo"
    src="http://localhot/algo.jpg"
  />
  <p>
    Prueba titulo
  </p>
</div>
`;
